build.sbt

```diff
@@ -1,13 +1,27 @@
 lazy val root = (project in file("."))
   .settings(basicSettings: _*)
+  .settings(
+    libraryDependencies ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, n)) if n <= 12 =>
+	  List(compilerPlugin("org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full))
```

Comment:
Be careful with paradise `v2.1.1`, which is not for all scala `12.x`.  `2.1.0` maybe the better choice.

Reply:
ok, I'll revert back to `2.1.0`; wil set scala 2.12 to `2.12.10` since this is the last version supported by the plugin.